### PR TITLE
Mark `.{1..9}` as generic file-extensions

### PR DESCRIPTION
**Pull request by Alhadis**
## Description
Successor of #4258, which makes use of Linguist's new [generic extensions](https://github.com/github/linguist/pull/4936) feature. To recap:

> Many files on GitHub receive an incorrect classification of "Roff", simply because their filenames end in a numeric suffix: 
> 
> ```
> Gemfile-3.0.3
> GFDL-1.2
> AFL-1.2
> mkfontscale-1.1.1
> ```
> 
> Change-logs and version-numbers are particularly problematic examples of these. The [`ValveSoftware/steam-runtime`](https://github.com/ValveSoftware/steam-runtime), for instance, was shown as 72.8% Roff (until [I fixed it](https://github.com/ValveSoftware/steam-runtime/pull/143)), because of [three license files](https://github.com/ValveSoftware/steam-runtime/search?l=roff) with version-numbers. [`haskell-CI/haskell-ci`](https://github.com/haskell-CI/haskell-ci/search?l=roff) was another example (until [fixed](haskell-CI/haskell-ci#223)). You can find those and more by checking GitHub's list of [trending Roff repositories](https://github.com/trending/roff?since=daily).

## Checklist:
- [x] **I am adding new or changing current functionality**
	- [x] I have added or updated the tests for the new or changed functionality.

### CODE REVIEW

**Comment by Alhadis** [comment]
@lildude Once this gets merged, and with your blessing, I'll follow-up with a PR to move the generic extensions list to its own file (`lib/linguist/generic.yml`).

File: test/fixtures/Generic/7/nil/samurai.7 [hunk 1]
```diff
@@ -0,0 +1,7 @@
+菊千代
```
Comment (smola) [inline test/fixtures/Generic/7/nil/samurai.7:1]: Nice easter eggs here :smile:

**Review by smola — APPROVED** [review]

**Review by lildude — APPROVED** [review]
LGTM

**Comment by lildude** [comment]
> Once this gets merged, and with your blessing, I'll follow-up with a PR to move the generic extensions list to its own file (`lib/linguist/generic.yml`).

Go for it. 🙇‍♂️

**Comment by Alhadis** [comment]
Done. 👍 See #5070.

**Pull request merged**